### ISSUE TITLE
Tab should auto-indent to the correct depth

### ISSUE THREAD

**Issue by nathansobo** [zed-industries/feedback#175]
If you paste the following into Zed and place your cursor at column 0 of the empty line, hitting `tab` doesn't indent you to the correct depth. For deeply nested files this is pretty annoying.

```js
if (true) {
  if (false) {
    if (true) {
  
    }
  }
}
```

**Comment by jansol** [comment]
FWIW I'd rather it didn't try to guess what the correct depth is. Many editors try, and with some code bases I work with they reliably guess wrong. (For example one has different indent rules for C and C++ files for historical reasons. I don't need my editor getting confused and making it more annoying to work with than it already is.)

Another common case for me to be on an empty line and press tab is when I am placing the closing brace(s). Usually I have to press tab, and then backspace since the editor is sure I want to indent to the level of the block's content. Some editors are even more "helpful": if I press backspace at the end of a line that only contains white space, they clear out the entire line instead of just the last character or last level of indent.

TL;DR: please at least leave this user configurable
